File: src/main/java/com/wezaam/withdrawal/model/WithdrawalScheduled.java
package com.wezaam.withdrawal.model;

import com.wezaam.withdrawal.utils.Utils;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import java.time.Instant;


@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "scheduled_withdrawals")
public class WithdrawalScheduled extends Withdrawal {
    @Getter
    @Setter
    private Instant executeAt;

    //Alternative setter, used by ModelMapper
    public void setExecuteAt(String executeAt) {
        this.executeAt = Utils.parseStringTimeToInstant(executeAt);
    }
}

File: src/main/java/com/wezaam/withdrawal/model/WithdrawalScheduled.kt
package com.wezaam.withdrawal.model;

import java.time.Instant;
import javax.persistence.*

@Entity(name = "scheduled_withdrawals")
data class WithdrawalScheduled(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id:Long,
    val transactionId:Long,
    val amount:Double,
    val createdAt: Instant,
    val executeAt: Instant,
    val userId:Long,
    @ManyToOne
    val paymentMethod:PaymentMethod,
    @Enumerated(EnumType.STRING)
    val status:WithdrawalStatus
)
